src/commands/basemaps-github/github.ts

```diff
@@ -0,0 +1,116 @@
+import { Env, LogType } from '@basemaps/shared';
+import { execFileSync } from 'child_process';
+import { Octokit } from '@octokit/core';
+import { Api } from '@octokit/plugin-rest-endpoint-methods/dist-types/types.js';
+import { restEndpointMethods } from '@octokit/plugin-rest-endpoint-methods';
+
+export class Github {
+  repo: string;
+  org: string;
+  logger: LogType;
+  repoName: string;
+  octokit: Api;
+
+  constructor(repo: string, org: string, repoName: string, logger: LogType) {
+    this.repo = repo;
+    this.logger = logger;
+    this.org = org;
+    this.repoName = repoName;
+
+    const token = Env.get(Env.GitHubToken);
+    if (token == null) throw new Error('Please set up github token environment variable.');
+    this.octokit = restEndpointMethods(new Octokit({ auth: token }));
+  }
+
+  isOk = (s: number): boolean => s >= 200 && s <= 299;
+
+  /**
+   * Clone the repository
+   *
+   */
+  clone(): void {
+    const ssh = `git@github.com:${this.repo}.git`;
+    this.logger.info({ repository: this.repo }, 'GitHub: Clone');
+    execFileSync('git', ['clone', ssh]).toString().trim();
+  }
+
+  /**
+   * Get branch by name if exists, or create a new branch by name.
+   *
+   * @returns {branch} github references or the new created branch
+   */
+  getBranch(branch: string): string {
+    this.logger.info({ branch }, 'GitHub: Get branch');
+    try {
+      execFileSync('git', ['checkout', branch], { cwd: this.repoName }).toString().trim();
+      this.logger.info({ branch }, 'GitHub: Branch Checkout');
+      return branch;
+    } catch {
+      this.logger.info({ branch }, 'GitHub: Create New Branch');
+      execFileSync('git', ['checkout', '-b', branch], { cwd: this.repoName }).toString().trim();
+      return branch;
+    }
+  }
+
+  /**
+   * Config github user email and user name
+   *
+   */
+  configUser(): void {
```

Comment:
could we make this `configUser(userName, email)` and move the Env logic into the callers?

Reply:
I will totally remove this once this merged as we decided to use Github API.